tasks/Task.cpp

```diff
@@ -203,10 +203,13 @@ void Task::cleanupHook()
             canbus::CanOutputPort const& outputport(outputports[i]);
```

Comment:
As @pierrewillenbrockdfki has commented on a specific commit:

- you do not need the inner loop anymore (and do not need the outputport attribute), which was where the whole simplification was
- iterate-and-erase with C++ containers is tricky. erase invalidates the iterator (i.e. you can't do ++it on an erased iterator). The pattern is instead

~~~
for (... it = begin; it != end; ) {
   if (should_erase) {
       it=container.erase(it);
   }
   else {
       ++it;
   }
}
~~~